src/main/java/org/apache/cassandra/sidecar/cdc/CDCBookmark.java

```diff
@@ -0,0 +1,213 @@
+
+            CommitLogPosition.CommitLogPositionSerializer serializer =
+                    new CommitLogPosition.CommitLogPositionSerializer();
+
+            // TODO: JSON ser-de and write-rename instead of writing directly to the bookmark
+            try (FileOutputStream fileOutputStream = new FileOutputStream(
+                    new File(this.getBookmarkPath())))
+            {
+                DataOutputPlus outBuffer = new DataOutputBuffer();
+                serializer.serialize(lastPosition, outBuffer);
+                fileOutputStream.write(((DataOutputBuffer) outBuffer).getData());
+                fileOutputStream.flush();
+                this.lastSyncedPosition = lastPosition;
+                logger.info("Successfully synced bookmark {} to the file {}", this.lastSyncedPosition.toString(),
+                        this.getBookmarkPath());
+            }
+            catch (IOException e)
+            {
+                logger.error("Error when writing bookmark {} to the file {}", lastPosition.toString(),
+                        this.getBookmarkPath());
+            }
+        }
+        catch (Exception ex)
+        {
+            logger.error("Sync exception {}", ex.getMessage());
+        }
+    }
+
+    /**
+     * Gets the path to the CDC reader bookmark.
+     *
+     * @return complete path to the bookmark file.
+     */
+    public String getBookmarkPath()
+    {
+        return String.format("%s/%s", DatabaseDescriptor.getCDCLogLocation(),
+                BOOKMARK);
+    }
+
+    @Override
+    public void run()
+    {
+        this.syncBookmark();
+    }
+
+    /**
+     * Gets the last successfully processed commit log offset.
+     * This method is thread safe.
+     *
+     * @return last successfully processed commit log offset.
+     */
+    public CommitLogPosition getLastProcessedPosition()
+    {
+        CommitLogPosition lastPosition = null;
+        try
+        {
+            bookmarkLock.lock();
+            if (this.lastProcessedPosition != null)
+            {
+                lastPosition = new CommitLogPosition(this.lastProcessedPosition.segmentId,
+                        this.lastProcessedPosition.position);
+
+            }
+        }
+        finally
+        {
+            bookmarkLock.unlock();
+        }
+        return lastPosition;
+    }
+
+    /**
+     * Sets the last successfully processed commit log offset.
+     * This method is thread safe.
+     *
+     */
+    public void setLastProcessedPosition(CommitLogPosition processedPosition)
+    {
+        try
+        {
+            bookmarkLock.lock();
+            this.lastProcessedPosition = processedPosition;
+        }
+        finally
+        {
+            bookmarkLock.unlock();
+        }
+    }
+
+    /**
+     * Starts the background thread to write processed commit log positions to the disk.
+     * */
+    public void startBookmarkSync()
+    {
+        timer.schedule(this, 0, DatabaseDescriptor.getCommitLogSyncPeriod());
```

Comment:
I do not know what version of jdk do you use. for me ,I found that 
the Timer class with the schedule function can not set the delay input argument
 to be less that 1. must be bigger that 0. 
Or an exception will be throw